src/engines/nwn2/script/functions_object.cpp

```diff
@@ -384,6 +385,24 @@ void Functions::getMaxHitPoints(Aurora::NWScript::FunctionContext &ctx) {
 	}
 }
 
+void Functions::getIsTrapped(Aurora::NWScript::FunctionContext &ctx) {
+	NWN2::Object *object = NWN2::ObjectContainer::toObject(getParamObject(ctx, 0));
```

Comment:
Since now Doors and Placeables *are* traps, you could just use toTrigger() directly here, no? No need to check the type and handle each type separately, just toTrigger(), return if it isn't a trap, getIsTripped() if it is a trap.

Reply:
I thought about doing that, but wasn't sure whether or not you'd prefer having things coded out in detail. I'll go ahead and make the change this weekend.